python/ray/experimental/serve/global_state.py

```diff
@@ -1,107 +1,134 @@
-import time
-from collections import defaultdict, deque
-
-import requests
-
 import ray
-from ray.experimental.serve.kv_store_service import KVStoreProxyActor
-from ray.experimental.serve.queues import CentralizedQueuesActor
-from ray.experimental.serve.utils import logger
-from ray.experimental.serve.server import HTTPActor
+from ray.experimental.serve.constants import (
+    BOOTSTRAP_KV_STORE_CONN_KEY, DEFAULT_HTTP_HOST, DEFAULT_HTTP_PORT,
+    SERVE_NURSERY_NAME)
+from ray.experimental.serve.kv_store_service import (
+    BackendTable, RoutingTable, TrafficPolicyTable)
 from ray.experimental.serve.metric import (MetricMonitor,
                                            start_metric_monitor_loop)
+from ray.experimental.serve.queues import CentralizedQueuesActor
+from ray.experimental.serve.server import HTTPActor
+
+
+def start_initial_state(kv_store_connector):
+    nursery_handle = ActorNursery.remote()
+    ray.experimental.register_actor(SERVE_NURSERY_NAME, nursery_handle)
+
+    ray.get(
+        nursery_handle.store_bootstrap_state.remote(
+            BOOTSTRAP_KV_STORE_CONN_KEY, kv_store_connector))
+    return nursery_handle
+
+
+@ray.remote
+class ActorNursery:
+    """Initialize and store all actor handles.
 
-# TODO(simon): Global state currently is designed to resides in the driver
-#     process. In the next iteration, we will move all mutable states into
-#     two actors: (1) namespaced key-value store backed by persistent store
-#     (2) actor supervisors holding all actor handles and is responsible
-#     for new actor instantiation and dead actor termination.
+    Note:
+        This actor is necessary because ray will destory actors when the
+        original actor handle goes out of scope (when driver exit). Therefore
+        we need to initialize and store actor handles in a seperate actor.
+    """
+
+    def __init__(self):
+        # Dict: Actor handles -> tag
+        self.actor_handles = dict()
+
+        self.bootstrap_state = dict()
+
+    def start_actor(self, actor_cls, init_args, tag):
+        """Start an actor and add it to the nursery"""
+        handle = actor_cls.remote(*init_args)
+        self.actor_handles[handle] = tag
+        return [handle]
+
+    def start_actor_with_creator(self, creator, tag):
+        handle = creator()
+        self.actor_handles[handle] = tag
+        return [handle]
 
-LOG_PREFIX = "[Global State] "
+    def get_all_handles(self):
+        return {tag: handle for handle, tag in self.actor_handles.items()}
+
+    def get_handle(self, actor_tag):
+        return [self.get_all_handles()[actor_tag]]
+
+    def remove_handle(self, actor_tag):
+        [handle] = self.get_handle(actor_tag)
+        self.actor_handles.pop(handle)
+        del handle
+
+    def store_bootstrap_state(self, key, value):
+        self.bootstrap_state[key] = value
+
+    def get_bootstrap_state_dict(self):
```

Comment:
`bootstrap_state` stores everything required to perform initial connection from worker.

Reply:
For now it's just a closure for connecting to database